contract/src/main/kotlin/io/provenance/scope/loan/utility/FundingDataValidation.kt

```diff
@@ -0,0 +1,106 @@
+                                orError "Disbursement account must have exactly one specific type",
+                        )
+                        disburseAccount.financial.takeIf { it.isSet() }?.also { financialAccount ->
+                            requireThat(
+                                financialAccount.id.isSet()
+                                    orError "Disbursement bank account ID must be set",
+                                (financialAccount.accountNumber.length in 4..17)
+                                    orError "Disbursement bank account must have a valid account number",
+                                (financialAccount.routingNumber.length == 9)
+                                    orError "Disbursement bank account must have a valid routing number",
+                            )
+                            financialAccount.movementList.requireThatEach(listIndices = false) { moneyMovement ->
+                                requireThat(
+                                    (moneyMovement.ach.isSet() xor moneyMovement.wire.isSet()) orError
+                                        "Disbursement bank account money movement must have exactly one specific type",
+                                    (
+                                        moneyMovement.ach.isNotSet() ||
+                                            moneyMovement.ach.accountType !in listOf(
+                                            ACH.AccountType.ACCOUNT_TYPE_UNKNOWN,
+                                            ACH.AccountType.UNRECOGNIZED,
+                                        )
+                                        ) orError
+                                        "Disbursement bank account for ACH must have a valid type",
+                                    (moneyMovement.wire.isNotSet() || moneyMovement.wire.accountAddress.isSet())
+                                        orError "Disbursement bank account for wires must have a valid address",
+                                    (moneyMovement.wire.isNotSet() || moneyMovement.wire.wireInstructions.isNotBlank())
+                                        orError "Disbursement bank account for wires must have valid wire instructions",
+                                    (
+                                        moneyMovement.wire.isNotSet() ||
+                                            moneyMovement.wire.swiftInstructions.isNotSet() ||
+                                            moneyMovement.wire.swiftInstructions.swiftId.isNotBlank()
+                                        ) orError "Disbursement bank account for wires must have valid SWIFT bank account ID",
+                                    (
+                                        moneyMovement.wire.isNotSet() ||
+                                            moneyMovement.wire.swiftInstructions.isNotSet() ||
+                                            moneyMovement.wire.swiftInstructions.swiftBankAddress.isSet()
```

Comment:
This simply checks that at least one of the fields in the address (e.g. any of `street1`, `street2`, `city`, `state`, etc.) is not an empty string. Is further validation desirable, e.g. require `street1`, `city`, `state`, and `zip`?

Reply:
I'm into the idea of requiring street1, city, state, and zip.